imblearn/utils/testing.py

```diff
@@ -120,3 +125,72 @@ def is_abstract(c):
     # itemgetter is used to ensure the sort does not extend to the 2nd item of
     # the tuple
     return sorted(set(estimators), key=itemgetter(0))
+
+
+@contextmanager
+def warns(expected_warning, message=None, match=None):
+    """
+    Assert that a code block/function call warns ``expected_warning``
+    and raise a failure exception otherwise.
+
+    If using Python 2.5 or above, you may use this function as a
```

Comment:
you need also to use a more numpydoc way using Parameters Returns and Examples.
I would add a narrative documentation in the user guide instead of the docstring.